warc_metadata_sidecar.py

```diff
@@ -0,0 +1,277 @@
+        level=logging.INFO,
+        format='%(asctime)s %(levelname)s %(message)s',
+    )
+    logging.getLogger(__name__)
+    logging.info('Logging WARC metadata record information for %s', warc_file)
+
+    # Create sidecar filename, adding 'meta' as extension.
+    new_file = os.path.basename(warc_file)
+    meta_file = re.sub(r'warc(\.gz)?$', 'warc.meta.gz', new_file)
+    logging.info('Creating sidecar %s', meta_file)
+    warc_file_path = os.path.join(archive_dir, meta_file)
+
+    # Open the sidecar file to write in the metadata, open the warc file to get each record.
+    with open(warc_file_path, 'ab') as output, open(warc_file, 'rb') as stream:
+        record_count = 0
+        total_records = 0
+        text_mime = 0
+        non_text = 0
+        fido = ExtendFido()
+
+        writer = WARCWriter(output, gzip=False)  # TODO: gzip will equal True
+        warc_info = create_warcinfo_payload(new_file, operator, publisher)
+        # Create warcinfo record and write it into sidecar.
+        warcinfo_record = writer.create_warcinfo_record(meta_file, warc_info)
+        writer.write_record(warcinfo_record)
+
+        for record in ArchiveIterator(stream):
+            total_records += 1
+            if record.rec_type not in ['response', 'resource']:
+                continue
+            if 'text/dns' in record.rec_headers.get_header('Content-Type'):
+                continue
+            payload = io.BytesIO(record.content_stream().read())
+            # The payload is how we find the important info. Skip record if empty.
+            if not payload.read(1):
+                continue
+            url = record.rec_headers.get_header('WARC-Target-URI')
+            record_date = record.rec_headers.get_header('WARC-Date')
+            warcinfo_id = record.rec_headers.get_header('WARC-Warcinfo-ID')
+            warcrecord_id = record.rec_headers.get_header('WARC-Record-ID')
+            # Define specific warc_headers to include in sidecar.
+            warc_dict = {'WARC-Date': record_date, 'WARC-Concurrent-ID': warcrecord_id}
+            if warcinfo_id:
+                warc_dict['WARC-Warcinfo-ID'] = warcinfo_id
+
+            print(url)
+            payload.seek(0)
+            mime_and_puid = find_mime_and_puid(fido, payload)
+            result_dict = {}
+            lang_cld = None
+            if mime_and_puid[0] is not {}:
+                if mime_and_puid[0].get('fido') and mime_and_puid[0].get('python-magic'):
+                    record_mime = "r'(" + mime_and_puid[0].get('fido') + "|" + mime_and_puid[0].get('python-magic') + ")'"
+                else:
+                    if mime_and_puid[0].get('fido'):
+                        record_mime = mime_and_puid[0].get('fido')
+                    else:
+                        record_mime = mime_and_puid[0].get('python-magic')
+                # Check that the format matches either mime type or just the one.
+                if TEXT_FORMAT_MIMES.search(record_mime):
```

Comment:
This seems more complicated than it needs to be...I'm not sure if you need to build a raw string that looks like a regular expression for `TEXT_FORMAT_MIMES` to test. Since with `TEXT_FORMAT_MIMES` we are looking for anything with "text" or "html" or "xml" in it, would it work to concatenate any mime values in the dictionary as a single string and then check that?
```suggestion
            if TEXT_FORMAT_MIMES.search(' '.join(mime_dict.values())):
```
I think that should work even if there is nothing in the mime dictionary.